include/seqan3/alignment/pairwise/align_result_selector.hpp

```diff
@@ -20,12 +20,36 @@
 #include <seqan3/core/algorithm/configuration.hpp>
 #include <seqan3/core/metafunction/basic.hpp>
 #include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/core/metafunction/transformation_trait_or.hpp>
 #include <seqan3/core/type_list.hpp>
+#include <seqan3/range/decorator/gap_decorator.hpp>
 #include <seqan3/std/ranges>
 
 namespace seqan3::detail
 {
 
+/*!\brief A helper class to define the alignment return type.
```

Comment:
This is a transformation trait. I am currently refactoring the documentation for those, so let's leave this comment until we made a decision on how to properly document them so you can adjust this before/afterwards.